roles/openshift_grafana/tasks/install_grafana.yaml

```diff
@@ -163,7 +163,7 @@
 
 - name: set facts
   set_fact:
-    payload_data: "{{ grafana_datasource_json | regex_replace('grafana_name', grafana_datasource_name ) | regex_replace('prometheus_url', prometheus_route.results[0].spec.host ) | regex_replace('satoken',  prometheus_sa_token.stdout ) }}"
+    payload_data: "{{ grafana_datasource_json | regex_replace('grafana_name', grafana_datasource_name ) | regex_replace('prometheus_url', prometheus_route.results[0].spec.host ) | regex_replace('Bearer',  'Bearer ' + prometheus_sa_token.stdout) }}"
```

Comment:
I don’t think it’s necessary to go through the Internet here, I think just the Kubernetes Service that selects the Prometheus pods is good enough.

Reply:
we've discused it while ago and than we've decided to stay with the approach see also:
https://github.com/openshift/openshift-ansible/issues/7735

Reply:
That discussion seems to be about exposing Prometheus unauthenticated, which is not what I’m talking about.

The setup as I can tell is:

Grafana -> Internet -> OpenShift Route (prometheus-k8s) -> Kubernetes Service (prometheus-k8s.openshift-monitoring.svc) -> Prometheus Pod -> oauth proxy -> Prometheus container

The authentication part happens within the prometheus container, so going directly against the Prometheus pod would just save the round trip through the Internet, the authentication is the same.

Reply:
exactly, but not only the authentication, grafana quires prometheus thorough http, that means they have to bypass the oauth proxy ?!

Reply:
No, if you request the Kubernetes Service the oauth authentication is still necessary as oauth proxy is part of the prometheus container and the prometheus container listens only on localhost. So the shorter but still authenticated flow would be:

Kubernetes Service (prometheus-k8s.openshift-monitoring.svc) -> Prometheus Pod -> oauth proxy -> Prometheus container

Reply:
ok, but this playbook role uses route to authenticate and integrate the grafana http traffic.
so it's kind of useless to change it, since cluster-monitoring implements the service method instead of the route method as introduced here.

I think we can keep it for now, to continure the 3.10 work.
when the cluster-monitoring should support grafana in the stack ?

Reply:
Fine by me as long as we can agree it's an interim state.

Reply:
ok, is one of the other cluster-monitoring service accounts has read permission only ?